judgels-frontends/raphael/src/routes/uriel/contests/ContestsPage/ContestsPage.tsx

```diff
@@ -5,37 +5,89 @@ import { parse } from 'query-string';
 
 import Pagination from 'components/Pagination/Pagination';
 import { Card } from 'components/Card/Card';
+import SearchBox from 'components/SearchBox/SearchBox';
 import { Contest, ContestCreateData, ContestsResponse } from 'modules/api/uriel/contest';
 
 import { LoadingContestCard } from '../ContestCard/LoadingContestCard';
 import { ContestCard } from '../ContestCard/ContestCard';
 import { ContestCreateDialog } from '../ContestCreateDialog/ContestCreateDialog';
 import { contestActions as injectedContestActions } from '../modules/contestActions';
 
+import './ContestsPage.css';
+
 export interface ContestsPageProps extends RouteComponentProps<{ name: string }> {
   onGetContests: (name?: string, page?: number) => Promise<ContestsResponse>;
   onCreateContest: (data: ContestCreateData) => Promise<Contest>;
 }
 
+interface ContestsFilter {
+  name?: string;
+}
+
 export interface ContestsPageState {
   response?: ContestsResponse;
+  filter?: ContestsFilter;
+  isSearchBoxLoading?: boolean;
+  isContestsListLoading?: boolean;
 }
 
 class ContestsPage extends React.Component<ContestsPageProps, ContestsPageState> {
   private static PAGE_SIZE = 20;
 
   state: ContestsPageState = {};
 
+  componentWillMount() {
+    const queries = parse(this.props.location.search);
+    const name = queries.name as string;
+
+    this.setState({ filter: { name }, isSearchBoxLoading: false, isContestsListLoading: true });
+  }
+
   render() {
     return (
       <Card title="Contests">
-        {this.renderCreateDialog()}
+        {this.renderHeader()}
         {this.renderContests()}
         {this.renderPagination()}
       </Card>
     );
   }
 
+  private renderHeader = () => {
+    return (
+      <>
+        <div className="content-card__section create-contest-button-inline">{this.renderCreateDialog()}</div>
+        <div className="content-card__section search-box-inline">{this.renderSearchBox()}</div>
+        <div className="clearfix" />
+        {this.renderSearchResultsBanner()}
+      </>
+    );
+  };
+
+  private renderSearchResultsBanner = () => {
+    const filter = this.state.filter || {};
```

Comment:
This snippet occurs way too many times in this PR. We can refactor this into a method, something like:

```
private getNameFilter = (state: ContestsPageState) => {
  if (!state.filter || !state.filter.name) {
    return '';
  }
  return state.filter.name;
}
```

(state needs to be a parameter because this should be called also for `prevFilter`)